packages/ts-twoslasher/src/index.ts

```diff
@@ -709,6 +709,7 @@ export function twoslasher(code: string, extension: string, options: TwoSlashOpt
   // We can't pass the ts.DiagnosticResult out directly (it can't be JSON.stringified)
   for (const err of relevantErrors) {
     const codeWhereErrorLives = env.sys.readFile(err.file!.fileName)!
+    const lineOffset = code.split("\n").indexOf(codeWhereErrorLives.split("\n")[0])
```

Comment:
this is a really naive way of doing it, I'll try to push a refactor later today

Reply:
👍🏻 structurally looks right though

Reply:
It works in most cases, but if two filenames happen to start with the same line, it will always pick the first one.

I was thinking of either:
- matching the `// filename: ` name which corresponds to `err.file!.fileName)`
- adding the offset info directly in `nameContent` (since that's where the code is being traverse and split) so that it can be referenced here.

Any preference or alternative approaches?

Reply:
Ok, I've pushed a refactor which implements the first proposal (so that the diff is more contained).

I've also added a test case which stresses the edge case I mentioned above (two virtual files beginning with the same line). The older implementation would yield and incorrect result, while the revised one works

Reply:
Yeah, this looks really solid to me, nice call 👍🏻